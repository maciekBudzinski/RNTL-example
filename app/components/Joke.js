import React, {useState, useEffect} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import axios from 'axios';

const useChuckNorrisJokeFetch = () => {
  const [joke, setJoke] = useState('');

  const getTextFromServer = async () => {
    const response = await axios.get('https://api.chucknorris.io/jokes/random');
    setJoke(response.data.value);
  };

  useEffect(() => {
    getTextFromServer();
  }, []);

  return joke;
};

const Joke = () => {
  const jokeText = useChuckNorrisJokeFetch();

  return (
    <View style={styles.wrapper}>
      <Text style={styles.text}>{jokeText}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  text: {
    fontSize: 22,
  },
});

export default Joke;
