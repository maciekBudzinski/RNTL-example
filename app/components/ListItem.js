import React, { useCallback, useEffect } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const ListItem = ({
  name,
  id,
  deleteItem,
  handleUnmount = () => console.log('unmount'),
  handleNameChange = itemName => console.log(itemName),
}) => {
  const handleDeletePress = useCallback(() => {
    deleteItem(id);
  }, [deleteItem, id]);

  useEffect(() => handleUnmount());

  useEffect(() => {
    handleNameChange(name);
  }, [handleNameChange, name]);

  return (
    <View testID="wrapper" style={styles.wrapper}>
      <Text style={styles.text}>{name}</Text>
      <View style={styles.buttonsContainer}>
        <TouchableOpacity onPress={handleDeletePress}>
          <Text style={styles.deleteItem}>Delete</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  text: {
    fontSize: 22,
  },
  buttonsContainer: {
    alignItems: 'center',
  },
  deleteItem: {
    color: 'red',
  },

});

export default ListItem;
