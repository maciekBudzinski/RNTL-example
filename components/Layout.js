import React from 'react';
import {View, SafeAreaView, StyleSheet} from 'react-native';

const Layout = ({children}) => (
  <SafeAreaView style={styles.safeArea}>
    <View style={styles.wrapper}>
      {children}
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
  wrapper: {
    padding: 16,
    flex: 1,
  },
});

export default Layout;
