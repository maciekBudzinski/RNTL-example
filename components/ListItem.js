import React, {useCallback} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

const ListItem = ({name, id, deleteItem}) => {
  const handleDeletePress = useCallback(()=> {
    deleteItem(id);
  },[deleteItem, id]);

  return (
    <View testID="wrapper" style={styles.wrapper}>
      <Text style={styles.text}>{name}</Text>
      <View style={styles.buttonsContainer}>
        <TouchableOpacity onPress={handleDeletePress}>
          <Text style={styles.deleteItem}>Delete</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  text: {
    fontSize: 22,
  },
  buttonsContainer: {
    alignItems: 'center',
  },
  deleteItem: {
    color: 'red',
  },

});

export default ListItem;
