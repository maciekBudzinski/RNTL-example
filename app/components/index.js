import Layout from './Layout';
import ListItem from './ListItem';
import Joke from './Joke';

export {
  Layout,
  ListItem,
  Joke,
};
