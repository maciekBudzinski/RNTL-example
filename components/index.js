import Layout from './Layout';
import ListItem from './ListItem';

export {
  Layout,
  ListItem,
};
