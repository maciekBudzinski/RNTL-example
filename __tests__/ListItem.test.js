import React from 'react';
import { TouchableOpacity } from 'react-native';
import { ListItem } from 'components';
import { render, fireEvent } from 'react-native-testing-library';

describe('Components: ListItem', () => {
  const defaultProps = {
    id: 12345,
    name: 'Todo item',
    deleteItem: jest.fn(),
  };
  it('Should match snapshot', () => {
    const wrapper = render(<ListItem {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render delete button', () => {
    const wrapper = render(<ListItem {...defaultProps} />);
    const deleteText = wrapper.getByText('Delete');
    expect(deleteText).toBeTruthy();
  });

  it('Should render red delete button', () => {
    const wrapper = render(<ListItem {...defaultProps} />);

    const deleteText = wrapper.getByText('Delete');
    const deleteTextStyle = deleteText.props.style;

    expect(deleteTextStyle.color).toBe('red');
  });


  it('Should call deleteItem function with key parameter', () => {
    const wrapper = render(<ListItem {...defaultProps} />);
    const button = wrapper.getByType(TouchableOpacity);
    fireEvent.press(button);
    expect(defaultProps.deleteItem).toHaveBeenCalledTimes(1);
    expect(defaultProps.deleteItem).toHaveBeenCalledWith(defaultProps.id);
  });
});

it('should call unmount function', ()=> {
  const wrapper = render(<ListItem {...defaultProps} />);
  wrapper.unmount();
  expect(defaultProps.handleUnmount).toHaveBeenCalledTimes(1);
});


//update
it.only('should call handleNameChange', ()=>{
  const wrapper = render(<ListItem {...defaultProps} />);
  wrapper.rerender(<ListItem {...defaultProps} name={'new name'} />);
  expect(defaultProps.handleNameChange).toHaveBeenCalledTimes(2);
});

