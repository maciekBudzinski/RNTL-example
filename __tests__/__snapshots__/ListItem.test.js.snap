// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: ListItem Should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
  >
    Todo item
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
          }
        }
      >
        Delete
      </Text>
    </View>
  </View>
</View>
`;
